<a href="https://colab.research.google.com/github/sordenesf/predicciones-de-ventas/blob/main/Proyecto_de_ventas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Se parte montando unidad de Drive para la carga de la data

In [737]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [738]:
import pandas as pd

file = '/content/drive/MyDrive/Coding Dojo/Week 1/sales_predictions_2023.csv'
df = pd.read_csv(file)

Se muestran las 10 primeras filas del dataframe

In [739]:
df.head(10)

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15        9.300          Low Fat         0.016047   
1           DRC01        5.920          Regular         0.019278   
2           FDN15       17.500          Low Fat         0.016760   
3           FDX07       19.200          Regular         0.000000   
4           NCD19        8.930          Low Fat         0.000000   
5           FDP36       10.395          Regular         0.000000   
6           FDO10       13.650          Regular         0.012741   
7           FDP10          NaN          Low Fat         0.127470   
8           FDH17       16.200          Regular         0.016687   
9           FDU28       19.200          Regular         0.094450   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   
5           Baking Goods   51.4008            OUT018   
6            Snack Foods   57.6588            OUT013   
7            Snack Foods  107.7622            OUT027   
8           Frozen Foods   96.9726            OUT045   
9           Frozen Foods  187.8214            OUT017   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   
5                       2009      Medium               Tier 3   
6                       1987        High               Tier 3   
7                       1985      Medium               Tier 3   
8                       2002         NaN               Tier 2   
9                       2007         NaN               Tier 2   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052  
5  Supermarket Type2           556.6088  
6  Supermarket Type1           343.5528  
7  Supermarket Type3          4022.7636  
8  Supermarket Type1          1076.5986  
9  Supermarket Type1          4710.5350

Se muestran las últimas 10 filas del dataframe

In [740]:
df.tail(50)

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
8473           DRI47       14.700          Low Fat         0.035016   
8474           NCJ42       19.750          Low Fat         0.014330   
8475           NCS17       18.600          Low Fat         0.080627   
8476           NCO41       12.500          Low Fat         0.018887   
8477           FDC05       13.100          Regular         0.099343   
8478           FDX34        6.195          Low Fat         0.071972   
8479           FDL10        8.395          Low Fat         0.039554   
8480           FDQ58          NaN          Low Fat         0.000000   
8481           FDT56       16.000          Regular         0.115827   
8482           FDZ28       20.000          Regular         0.051702   
8483           DRI11        8.260          Low Fat         0.034474   
8484           DRJ49        6.865          Low Fat         0.000000   
8485           DRK37          NaN          Low Fat         0.043792   
8486           FDR20       20.000          Regular         0.000000   
8487           DRG13          NaN          Low Fat         0.037006   
8488           NCN14          NaN          Low Fat         0.091473   
8489           FDV13       17.350          Regular         0.027723   
8490           FDU44          NaN          Regular         0.102296   
8491           FDO03       10.395          Regular         0.037033   
8492           FDT34        9.300          Low Fat         0.174350   
8493           FDP21        7.420          Regular         0.025886   
8494           NCI54       15.200          Low Fat         0.000000   
8495           FDE22        9.695          Low Fat         0.029567   
8496           FDJ57        7.420          Regular         0.021696   
8497           FDT08       13.650          Low Fat         0.049209   
8498           NCP54       15.350          Low Fat         0.035293   
8499           NCK53       11.600          Low Fat         0.037574   
8500           NCQ42       20.350          Low Fat         0.000000   
8501           FDW21        5.340          Regular         0.005998   
8502           NCH43        8.420          Low Fat         0.070712   
8503           FDQ44       20.500          Low Fat         0.036133   
8504           NCN18          NaN          Low Fat         0.124111   
8505           FDB46       10.500          Regular         0.094146   
8506           DRF37       17.250          Low Fat         0.084676   
8507           FDN28        5.880          Regular         0.030242   
8508           FDW31       11.350          Regular         0.043246   
8509           FDG45        8.100          Low Fat         0.214306   
8510           FDN58       13.800          Regular         0.056862   
8511           FDF05       17.500          Low Fat         0.026980   
8512           FDR26       20.700          Low Fat         0.042801   
8513           FDH31       12.000          Regular         0.020407   
8514           FDA01       15.000          Regular         0.054489   
8515           FDH24       20.700          Low Fat         0.021518   
8516           NCJ19       18.600          Low Fat         0.118661   
8517           FDF53       20.750              reg         0.083607   
8518           FDF22        6.865          Low Fat         0.056783   
8519           FDS36        8.380          Regular         0.046982   
8520           NCJ29       10.600          Low Fat         0.035186   
8521           FDN46        7.210          Regular         0.145221   
8522           DRG01       14.800          Low Fat         0.044878   

                  Item_Type  Item_MRP Outlet_Identifier  \
8473            Hard Drinks  144.3128            OUT010   
8474              Household  104.2332            OUT045   
8475     Health and Hygiene   92.5436            OUT049   
8476     Health and Hygiene   98.8384            OUT045   
8477           Frozen Foods  198.1768            OUT017   
8478            Snack Foods  

#Limpieza de datos

1) ¿Cuántas filas y columnas hay?

In [741]:
df.shape

(8523, 12)

2) ¿Cuáles son los tipos de datos de cada variable?

In [742]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


3) ¿Hay duplicados? Si es el caso, elimínelos.

In [743]:
df.duplicated().sum()

0

No se identifican duplicados, por lo tanto, no se hace tratamiento para este tipo de situación

4) Identifique los valores faltantes.

Total de valores faltantes por cada una de las columnas del dataframe:

In [744]:

df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

El porcentaje de los valores faltantes respecto al total de cada columna es el siguiente:

In [745]:
df.isna().sum()*100/df.shape[0]

Item_Identifier               0.000000
Item_Weight                  17.165317
Item_Fat_Content              0.000000
Item_Visibility               0.000000
Item_Type                     0.000000
Item_MRP                      0.000000
Outlet_Identifier             0.000000
Outlet_Establishment_Year     0.000000
Outlet_Size                  28.276428
Outlet_Location_Type          0.000000
Outlet_Type                   0.000000
Item_Outlet_Sales             0.000000
dtype: float64

Se detectan datos faltantes en las columnas 'Item_Weight' y en 'Outlet_Size'. En el caso de la columna 'Item_Weight', los datos son de tipo float64 y corresonden a un 17,16% del total de la columnas. Por otra parte, 'Outlet_Size' tien un 28,27% de datos faltantes y que son del tipo object

5) Decida qué hacer con los valores faltantes y cómo hacerlo. (Esto requiere de su criterio, así que explique cada elección).

Debido a la gran porcentaje de datos faltantes (superior al 2%), se buscará la forma de completar los datos en virtud del resto de los datos. Para ello, primero se realizará el análisis de la variable 'Outlet_Size que es del tipo categórica:

Se crea un filtro de los datos nulos en de la columna 'Oulet_size' para filtrar los datos nulos cuando sea pertinente:

In [746]:
filtro_na_outlet = df['Outlet_Size'].isna()

Según la documentación, la columna Oulet_Size está asocida a una entidad superior (Oulet), por lo cual, se seleccionan otras columnas del dataframe que estén relacionadas con Oulets. Esto con la finalidad de buscar posibles relaciones:

In [747]:
outlet_columns = ['Outlet_Identifier','Outlet_Establishment_Year', 'Outlet_Location_Type', 'Outlet_Type' ]

A continuación, para cada columna relacionada con Outlet, se agruparan los datos nulos (en porcentaje respecto al total de registros):

Datos nulos en variable 'Outlet_Identifier':

In [748]:
df[filtro_na_outlet].groupby(['Outlet_Identifier'])['Item_Outlet_Sales'].count()*100/df.shape[0]

Outlet_Identifier
OUT010     6.511792
OUT017    10.864719
OUT045    10.899918
Name: Item_Outlet_Sales, dtype: float64

Datos nulos en variable 'Outlet_Establishment_Year':

In [749]:
df[filtro_na_outlet].groupby(['Outlet_Establishment_Year'])['Item_Outlet_Sales'].count()*100/df.shape[0]

Outlet_Establishment_Year
1998     6.511792
2002    10.899918
2007    10.864719
Name: Item_Outlet_Sales, dtype: float64

Datos nulos en variable 'Outlet_Location_Type':

In [750]:
df[filtro_na_outlet].groupby(['Outlet_Location_Type'])['Item_Outlet_Sales'].count()*100/df.shape[0]

Outlet_Location_Type
Tier 2    21.764637
Tier 3     6.511792
Name: Item_Outlet_Sales, dtype: float64

Datos nulos en variable 'Outlet_Type':

In [751]:
df[filtro_na_outlet].groupby(['Outlet_Type'])['Item_Outlet_Sales'].count()*100/df.shape[0]

Outlet_Type
Grocery Store         6.511792
Supermarket Type1    21.764637
Name: Item_Outlet_Sales, dtype: float64

Se observa que en las columnas 'Outlet_Identifier' y 'Outlet_Establishment_Year', los porcetajes se encuentran en dos categorías y son los mismos en ambas columnas. En forma análoga, los porcentajes en 'Outlet_Location_Type' y en 'Outlet_Type' se dividen en 3 categorías y en las dos columnas son los mismos. Con esta información, se sospecha que los datos nulos se encuentran en unas pocas categorías.

A continuación, se agrupan (suma de Item_Outlet_Sales) los datos ocupando las 4 columnas 'Outlet' que no tienen datos nulos:

In [752]:
df[filtro_na_outlet].groupby(outlet_columns)[['Item_Outlet_Sales']].sum().round(0)

Item_Outlet_Sales
Outlet_Identifier Outlet_Establishment_Year Outlet_Location_Type Outlet_Type                         
OUT010            1998                      Tier 3               Grocery Store               188340.0
OUT017            2007                      Tier 2               Supermarket Type1          2167465.0
OUT045            2002                      Tier 2               Supermarket Type1          2036725.0

De la tabla anterior, se obseva que los datos nulos pertenecen específicamente a 3 Outlets, identificados como OUT010, OUT017 y OUT045.

Se suman los registros que **no tienen** datos faltantes en la columna 'Outlet_Size'

In [753]:
outlet_columns.append('Outlet_Size')

In [754]:
df.groupby(outlet_columns)[['Item_Outlet_Sales']].sum().round(0)

Item_Outlet_Sales
Outlet_Identifier Outlet_Establishment_Year Outlet_Location_Type Outlet_Type       Outlet_Size                   
OUT013            1987                      Tier 3               Supermarket Type1 High                 2142664.0
OUT018            2009                      Tier 3               Supermarket Type2 Medium               1851823.0
OUT019            1985                      Tier 1               Grocery Store     Small                 179694.0
OUT027            1985                      Tier 3               Supermarket Type3 Medium               3453926.0
OUT035            2004                      Tier 2               Supermarket Type1 Small                2268123.0
OUT046            1997                      Tier 1               Supermarket Type1 Small                2118395.0
OUT049            1999                      Tier 1               Supermarket Type1 Medium               2183970.0

En base a las tablas anteriores, se aplicarán los siguientes criterios para completar valores faltantes en la columna 'Outlet_Size':

- No se evidencias asociaciones relevantes de las columnas 'Outlet_Identifier' y 'Outlet_Establishment_Year' con respecto al tamaño del outlet, motivo por el que los criterios serán establecidos principalmente con el resto de las columnas

- De todos los tipos de outlets, el outlet OUT013 es el único de tipo 'Grocery Store', siendo de tamaño Small. Además, sus ventas son bastante bajas comparadas a los otros outlets. Ambas condiciones se cumplen en el outlet OUT010, por ende, será considerado tipo Small

- Los outlets OUT017 y OUT045 son tipo 'Supermarket Type1' y ubicados en 'Tier 2', con ventas muy similares. De los outlets que no tienen nulos, solo uno (OUT035) está localizado en 'Tier 2' y es de tipo Small.

- Complementando el punto anterior, si analizamos la categoría 'Outlet_Type', cuando el outlet es tipo 'Supermarket Type1' se pueden encontrar de los 3 tamaños. No obstante, si miramos las ventas, vemos que las ventas de OUT017 y OUT045 son similares a OUT046 (tamaño Small), además de ser inaugurados en fechas cercanas.  

- Teniendo como base los 2 puntos anteriores, OUT017 y OUT045 será considerados Small. A pesar de tener cierta incertidumbre en esta asignación, se prefiere introducir cierto error en la columna 'Outlet_Size' a perder más de un 20% de los registros.

In [755]:
df.groupby(['Outlet_Location_Type','Outlet_Size','Outlet_Identifier'])[['Item_Outlet_Sales']].sum().round(0)

Item_Outlet_Sales
Outlet_Location_Type Outlet_Size Outlet_Identifier                   
Tier 1               Medium      OUT049                     2183970.0
                     Small       OUT019                      179694.0
                                 OUT046                     2118395.0
Tier 2               Small       OUT035                     2268123.0
Tier 3               High        OUT013                     2142664.0
                     Medium      OUT018                     1851823.0
                                 OUT027                     3453926.0

In [756]:
resumen= df.groupby(['Outlet_Type','Outlet_Size','Outlet_Identifier'])[['Item_Outlet_Sales']].sum().round(0)
resumen.sort_values(by='Item_Outlet_Sales', ascending=True)

Item_Outlet_Sales
Outlet_Type       Outlet_Size Outlet_Identifier                   
Grocery Store     Small       OUT019                      179694.0
Supermarket Type2 Medium      OUT018                     1851823.0
Supermarket Type1 Small       OUT046                     2118395.0
                  High        OUT013                     2142664.0
                  Medium      OUT049                     2183970.0
                  Small       OUT035                     2268123.0
Supermarket Type3 Medium      OUT027                     3453926.0

Se realizan los cambios

In [757]:
df['Outlet_Size']= df['Outlet_Size'].fillna('Small')
df.groupby(['Outlet_Identifier','Outlet_Size'])[['Item_Outlet_Sales']].sum().round(0)


Item_Outlet_Sales
Outlet_Identifier Outlet_Size                   
OUT010            Small                 188340.0
OUT013            High                 2142664.0
OUT017            Small                2167465.0
OUT018            Medium               1851823.0
OUT019            Small                 179694.0
OUT027            Medium               3453926.0
OUT035            Small                2268123.0
OUT045            Small                2036725.0
OUT046            Small                2118395.0
OUT049            Medium               2183970.0

In [758]:
df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                     0
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

Ahora, se analizarán los datos de la columna 'Item_Weight'

Se crea un filtro de valores nulos en la columna 'Item_Weight'

In [759]:
filtro_na_weight = df['Item_Weight'].isna()


Se detecta que 'Item_Fat_Content' tiene datos inconsistentes, por lo que se tratan antes de continuar

In [760]:
df['Item_Fat_Content'].value_counts()

Item_Fat_Content
Low Fat    5089
Regular    2889
LF          316
reg         117
low fat     112
Name: count, dtype: int64

Se crea un diccionario para datos incosistentes:

In [761]:
dict_item_fat_content = {'LF':'Low Fat','reg':'Regular','low fat':'Low Fat'}

Se procede con el cambio y se verifican los valores:

In [762]:
df['Item_Fat_Content']= df['Item_Fat_Content'].replace(dict_item_fat_content)
df['Item_Fat_Content'].value_counts()

Item_Fat_Content
Low Fat    5517
Regular    3006
Name: count, dtype: int64

De la documentación, se sabe que las variables que se encuentran asociadas al peso de un producto son tipo del producto como también su contenido de grasa. El identificador del producto, la visibilidad del mismo y el precio de lista no afectan directamente al peso.

Agrupamos los datos por contenido de grasa y por tipo de producto, obteniendo estádisticas resumen mediante la variable 'Item_Weight':

In [763]:
df[filtro_na_weight].groupby(['Item_Identifier','Item_Type','Item_Fat_Content'])[['Item_Identifier']].agg('count')

Item_Identifier
Item_Identifier Item_Type          Item_Fat_Content                 
DRA24           Soft Drinks        Regular                         2
DRA59           Soft Drinks        Regular                         2
DRB01           Soft Drinks        Low Fat                         1
DRB25           Soft Drinks        Low Fat                         1
DRB48           Soft Drinks        Regular                         1
...                                                              ...
NCZ17           Health and Hygiene Low Fat                         1
NCZ18           Household          Low Fat                         1
NCZ30           Household          Low Fat                         1
NCZ53           Health and Hygiene Low Fat                         1
NCZ54           Household          Low Fat                         2

[1142 rows x 1 columns]

Existen 1142 productos que tienen al menos un registro sin su peso

Se buscan estádisticos generales de 'Item_Weight':

In [764]:
df[['Item_Weight']].agg(['min','max','median','mean','std'])

Item_Weight
min        4.555000
max       21.350000
median    12.600000
mean      12.857645
std        4.643456

Ahora, se determinan estadísticos del peso del producto, pero agrupados por por el contenido de grasa:

In [765]:
df.groupby('Item_Fat_Content')[['Item_Weight']].agg(['min','max','median','mean','std'])

Item_Weight                                   
                         min    max median       mean       std
Item_Fat_Content                                               
Low Fat                4.590  21.35  12.65  12.937387  4.653787
Regular                4.555  21.20  12.50  12.711654  4.621851

Por otra parte, el peso se agrupa mediante el tipo de producto

In [766]:
df.groupby('Item_Type')[['Item_Weight']].agg(['min','max','median','mean','std'])

Item_Weight                                    
                              min    max  median       mean       std
Item_Type                                                            
Baking Goods                4.880  20.85  11.650  12.277108  4.773622
Breads                      4.635  20.85  10.600  11.346936  4.440540
Breakfast                   6.425  21.10  10.695  12.768202  5.038131
Canned                      4.615  21.35  12.150  12.305705  4.586564
Dairy                       4.805  20.70  13.350  13.426069  4.686532
Frozen Foods                4.555  20.85  12.850  12.867061  4.507383
Fruits and Vegetables       5.460  21.35  13.100  13.224769  4.575275
Hard Drinks                 4.610  19.70  10.100  11.400328  4.239144
Health and Hygiene          5.175  21.25  12.150  13.142314  4.512313
Household                   5.030  21.25  13.150  13.384736  4.998845
Meat                        5.150  21.25  12.350  12.817344  4.670812
Others                      5.500  20.50  14.500  13.853285  4.225534
Seafood                     5.365  20.75  11.650  12.552843  5.473830
Snack Foods                 5.095  21.25  13.150  12.987880  4.531256
Soft Drinks                 4.590  20.75  11.800  11.847460  4.403699
Starchy Foods               6.695  21.20  13.175  13.690731  4.010061

Como se observa de las 3 tablas, se comenta lo siguiente:

- Los estadísticos al agrupar por contenido de grasa son prácticamente los mismos a los que se obtienen de la variable peso de forma general (sin usar groupby).

- Al agrupar por tipo de productos, cada uno de los categorías tiene variaciones respecto a los estísticos de la variable peso.

Por este motivo, se imputarán los valores nulo según el promedio de cada categoría de la variable 'Item_Type'

In [767]:
agrupacion = df.groupby('Item_Type')[['Item_Weight']].mean()
agrupacion

Item_Weight
Item_Type                         
Baking Goods             12.277108
Breads                   11.346936
Breakfast                12.768202
Canned                   12.305705
Dairy                    13.426069
Frozen Foods             12.867061
Fruits and Vegetables    13.224769
Hard Drinks              11.400328
Health and Hygiene       13.142314
Household                13.384736
Meat                     12.817344
Others                   13.853285
Seafood                  12.552843
Snack Foods              12.987880
Soft Drinks              11.847460
Starchy Foods            13.690731

Se genera un diccionario donde la llave corresponde al tipo de producto y el respectivo valor será el promedio obtenido

In [768]:
llaves = agrupacion.index
valores = agrupacion.values.ravel() #ravel para quitar una dimension del array
diccionario = dict(zip(llaves,valores))
diccionario

{'Baking Goods': 12.277108208955223,
 'Breads': 11.346936274509805,
 'Breakfast': 12.768202247191013,
 'Canned': 12.305705009276437,
 'Dairy': 13.42606890459364,
 'Frozen Foods': 12.867061281337046,
 'Fruits and Vegetables': 13.224769381746812,
 'Hard Drinks': 11.400327868852461,
 'Health and Hygiene': 13.14231395348837,
 'Household': 13.384736495388669,
 'Meat': 12.81734421364985,
 'Others': 13.853284671532847,
 'Seafood': 12.552843137254904,
 'Snack Foods': 12.987879554655871,
 'Soft Drinks': 11.847459893048129,
 'Starchy Foods': 13.69073076923077}

Ciclo for para asignar a cada valor nulo (incluidos en filtro_na_weight) el promedio del peso correspondiente a su categoría 'Item_Type'

In [769]:
for llave,valor in diccionario.items():
  filtro = df['Item_Type']== llave
  df.loc[filtro_na_weight & filtro, ['Item_Weight']] = df.loc[filtro_na_weight & filtro, ['Item_Weight']].fillna(valor)

In [770]:
df.tail(50)

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
8473           DRI47    14.700000          Low Fat         0.035016   
8474           NCJ42    19.750000          Low Fat         0.014330   
8475           NCS17    18.600000          Low Fat         0.080627   
8476           NCO41    12.500000          Low Fat         0.018887   
8477           FDC05    13.100000          Regular         0.099343   
8478           FDX34     6.195000          Low Fat         0.071972   
8479           FDL10     8.395000          Low Fat         0.039554   
8480           FDQ58    12.987880          Low Fat         0.000000   
8481           FDT56    16.000000          Regular         0.115827   
8482           FDZ28    20.000000          Regular         0.051702   
8483           DRI11     8.260000          Low Fat         0.034474   
8484           DRJ49     6.865000          Low Fat         0.000000   
8485           DRK37    11.847460          Low Fat         0.043792   
8486           FDR20    20.000000          Regular         0.000000   
8487           DRG13    11.847460          Low Fat         0.037006   
8488           NCN14    13.853285          Low Fat         0.091473   
8489           FDV13    17.350000          Regular         0.027723   
8490           FDU44    13.224769          Regular         0.102296   
8491           FDO03    10.395000          Regular         0.037033   
8492           FDT34     9.300000          Low Fat         0.174350   
8493           FDP21     7.420000          Regular         0.025886   
8494           NCI54    15.200000          Low Fat         0.000000   
8495           FDE22     9.695000          Low Fat         0.029567   
8496           FDJ57     7.420000          Regular         0.021696   
8497           FDT08    13.650000          Low Fat         0.049209   
8498           NCP54    15.350000          Low Fat         0.035293   
8499           NCK53    11.600000          Low Fat         0.037574   
8500           NCQ42    20.350000          Low Fat         0.000000   
8501           FDW21     5.340000          Regular         0.005998   
8502           NCH43     8.420000          Low Fat         0.070712   
8503           FDQ44    20.500000          Low Fat         0.036133   
8504           NCN18    13.384736          Low Fat         0.124111   
8505           FDB46    10.500000          Regular         0.094146   
8506           DRF37    17.250000          Low Fat         0.084676   
8507           FDN28     5.880000          Regular         0.030242   
8508           FDW31    11.350000          Regular         0.043246   
8509           FDG45     8.100000          Low Fat         0.214306   
8510           FDN58    13.800000          Regular         0.056862   
8511           FDF05    17.500000          Low Fat         0.026980   
8512           FDR26    20.700000          Low Fat         0.042801   
8513           FDH31    12.000000          Regular         0.020407   
8514           FDA01    15.000000          Regular         0.054489   
8515           FDH24    20.700000          Low Fat         0.021518   
8516           NCJ19    18.600000          Low Fat         0.118661   
8517           FDF53    20.750000          Regular         0.083607   
8518           FDF22     6.865000          Low Fat         0.056783   
8519           FDS36     8.380000          Regular         0.046982   
8520           NCJ29    10.600000          Low Fat         0.035186   
8521           FDN46     7.210000          Regular         0.145221   
8522           DRG01    14.800000          Low Fat         0.044878   

                  Item_Type  Item_MRP Outlet_Identifier  \
8473            Hard Drinks  144.3128            OUT010   
8474              Household  104.2332            OUT045   
8475     Health and Hygiene   92.5436            OUT049   
8476     Health and Hygiene   98.8384            OUT045   
8477           Frozen Foods  198.1768            OUT017   
8478            Snack Foods  

6) Confirme que no hay valores faltantes después de tratarlos.

In [771]:
df.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

7) Encuentre y arregle cualquier categoría inconsistente en los datos (ejemplo: arreglar los datos gato, Gato y Gatos de manera que sean consistentes)

En el punto 6) se encontró que 'Item_Far_Content' tenía datos inconsistentes, los cuales fueron abordados para el tratamiento de datos nulos. Se muestra que ahora hay solo 2 categorías:

In [772]:
df['Item_Fat_Content'].value_counts()

Item_Fat_Content
Low Fat    5517
Regular    3006
Name: count, dtype: int64

8) Para cualquier columna numérica, obtenga las estadísticas resumidas de cada una (mínimo, máximo y media)

In [773]:
df.describe().loc[['min','max','mean'],:]

Item_Weight  Item_Visibility    Item_MRP  Outlet_Establishment_Year  \
min       4.55500         0.000000   31.290000                1985.000000   
max      21.35000         0.328391  266.888400                2009.000000   
mean     12.85789         0.066132  140.992782                1997.831867   

      Item_Outlet_Sales  
min           33.290000  
max        13086.964800  
mean        2181.288914